src/mantine-hooks/src/use-state-validator/use-state-validator.ts

```diff
@@ -0,0 +1,18 @@
+import { useState } from 'react';
+
+export function useStateValidator<T>(
+  initialValue: T | null | undefined,
+  validator: (val: T | null | undefined) => boolean
+) {
+  let isValid = validator(initialValue);
+  const [valid, setValid] = useState<boolean>(isValid);
+  const [value, setValue] = useState<T | null | undefined>(isValid ? initialValue : undefined);
```

Comment:
initialValue should always be set, no matter of validation state, same in handleChange function